### ISSUE TITLE
Set up a docker for development

### ISSUE THREAD

**Issue by mirdaki** [ufosc/club-backend#3]
It should support Rust nightly (in order to run Rocket) as well as a PostgreSQL database that can be saved locally

**Comment by mirdaki** [comment]
Hey, @hjarrell so I'm running into a couple of problems I haven't been able to fix with Docker.

1. Docker compose doesn't seem to run the current code
    - I created a hello world program, which works with just a Dockerfile and docker compose 
    - I modify it to print 4, and the Dockerfile will work, but the docker compose still just prints hello world (I've brought it down and then up --build with no effect)
2. Something isn't being set up correctly with the database 
   - I get the following output when running docker compose
   ```
   db_1       | The files belonging to this database system will be owned by user "postgres".
   db_1       | This user must also own the server process.
   db_1       |
   db_1       | The database cluster will be initialized with locale "en_US.utf8".
   db_1       | The default database encoding has accordingly been set to "UTF8".
   db_1       | The default text search configuration will be set to "english".
   db_1       |
   db_1       | Data page checksums are disabled.
   db_1       |
   db_1       | fixing permissions on existing directory /var/lib/postgresql/data ... initdb: could not change permissions of directory "/var/lib/postgresql/data": Operation not permitted
   backend_1  | Hello, world!
   club-backend_db_1 exited with code 1
   club-backend_backend_1 exited with code 0
    ```
    - I've made sure the local db folder has 777 permissions, so my guess is it has something to do with that not happening on the docker container
    - I also suspect this might be why the I got the "db not found" error when running the example code

**Comment by hjarrell** [comment]
I'll take a look on Wednesday at casual coding!

**Comment by mirdaki** [comment]
Merged with #16
